pkg/server/commands/list_objects.go

```diff
@@ -118,7 +116,7 @@ func NewListObjectsQuery(ds storage.RelationshipTupleReader, opts ...ListObjects
 		resolveNodeLimit:        serverconfig.DefaultResolveNodeLimit,
 		resolveNodeBreadthLimit: serverconfig.DefaultResolveNodeBreadthLimit,
 		maxConcurrentReads:      serverconfig.DefaultMaxConcurrentReadsForListObjects,
-		checkOptions:            []graph.LocalCheckerOption{},
+		checkResolver:           checkResolver,
```

Comment:
Practical usages of this never allow for `checkResolver` to be nil. We always pass the server's check resolver, which is instantiated at server startup. See usages of `NewListObjectsQuery`. Adding a test for it and instantiating a default check resolver would solely be for testing purposes itself. What I'll do is change the signature for `NewListObjectsQuery` so that it returns an error, and if the datastore or check resolver are nil then we can return an error.

I think this raises a point though - commands should be internal packages, not part of `pkg/server`. A Go project shouldn't be importing the commands, they should be importing the Server and invoking the RPC handlers on it.

Reply:
I have changed the signature and added a unit test to account for the specific case where you pass in nil, but to the point above that shouldn't be exposed to the public anyways.